lib/ansible/modules/cloud/google/gcp_dns_resource_record_set.py

```diff
@@ -0,0 +1,428 @@
+      domain / subdomains direct to.
+      The record will include the domain/subdomain name, a type (i.e. A, AAA,
+      CAA, MX, CNAME, NS, etc)
+short_description: Creates a GCP ResourceRecordSet
+version_added: 2.6
+author: Google Inc. (@googlecloudplatform)
+requirements:
+    - python >= 2.6
+    - requests >= 2.18.4
+    - google-auth >= 1.3.0
+options:
+    state:
+        description:
+            - Whether the given object should exist in GCP
+        required: true
+        choices: ['present', 'absent']
+        default: 'present'
+    name:
+        description:
+            - For example, www.example.com.
+        required: true
+    type:
+        description:
+            - One of valid DNS resource types.
+        required: true
+    ttl:
+        description:
+            - Number of seconds that this ResourceRecordSet can be cached by
+              resolvers.
+        required: false
+    target:
+        description:
+            - As defined in RFC 1035 (section 5) and RFC 1034 (section 3.6.1)
+        required: false
+    managed_zone:
+        description:
+            - A reference to ManagedZone resource
+        required: true
+extends_documentation_fragment: gcp
+'''
+
+EXAMPLES = '''
+- name: create a managed zone
+  gcp_dns_managed_zone:
+      name: 'testzone-4-com'
+      dns_name: 'testzone-4.com.'
+      description: 'test zone'
+      project: "{{ gcp_project }}"
+      auth_kind: "{{ gcp_cred_kind }}"
+      service_account_file: "{{ gcp_cred_file }}"
+      scopes:
+        - https://www.googleapis.com/auth/ndev.clouddns.readwrite
+      state: present
+  register: managed_zone
+- name: create a resource record set
+  gcp_dns_resource_record_set:
+      name: 'www.testzone-4.com.'
+      managed_zone: "{{ managed_zone }}"
+      type: 'A'
+      ttl: 600
+      target:
+        - 10.1.2.3
+        - 40.5.6.7
+      project: testProject
+      auth_kind: service_account
+      service_account_file: /tmp/auth.pem
+      scopes:
+        - https://www.googleapis.com/auth/ndev.clouddns.readwrite
+      state: present
+'''
+
+RETURN = '''
+'''
+
+################################################################################
+# Imports
+################################################################################
+
+from ansible.module_utils.gcp_utils import navigate_hash, GcpSession, GcpModule, GcpRequestException
+import json
+import copy
+import datetime
+import time
+
+################################################################################
+# Main
+################################################################################
+
+
+def main():
+    """Main function"""
+
+    module = GcpModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['present', 'absent'], type='str'),
+            name=dict(required=True, type='str'),
+            type=dict(required=True, type='str'),
+            ttl=dict(type='int'),
```

Comment:
If this isn't provided, it should default to None but is still passed in the API request. Does the GCP API handle that?

Reply:
Yes, the GCP API will give you a error message.

Reply:
Should've provided better context on this:

GCP will ignore None for optional parameters. It will send an error message for required parameters.